test/test_bayes_net.py

```diff
@@ -186,13 +195,36 @@ def test_generate_household(self):
         self._check_household_generate(household_model)
 
     def test_load_bayes_net(self):
-        nodes = ('age', 'sex', 'income')
-        edges = {
-            'age': ['sex', 'income'],
-            'sex': ['income']
-        }
+        nodes = self._mock_nodes()
+        edges = self._mock_edges()
         structure = bayesnets.define_bayes_net_structure(nodes, edges)
         self.assertSequenceEqual(structure, ((), (0,), (0, 1)))
+        #self.assertSequenceEqual(structure, (frozenset([]), frozenset([0]), frozenset([0, 1])))
+
+    def test_train_bayes_net_from_generated_structure(self):
+        '''Ensure the structures created by define_bayes_net_structure 
+        are compatible with pomegranate
+        '''
+        nodes = self._mock_nodes()
+        edges = self._mock_edges()
+        person_structure = bayesnets.define_bayes_net_structure(nodes, edges)
+        person_training_data = self._mock_people_input()
+        training_data = bayesnets.SegmentedData.from_data(
+            person_training_data, self._person_fields(), segmenter=self._person_segmenter()
+        )
+        
+        self.assertEquals(
+            type(
+                BayesianNetworkModel.train(
+                    training_data,
+                    person_structure,
+                    self._person_fields,
+                    prior_data=None
+                )
+            ),
+            bayesnets.BayesianNetworkModel
```

Comment:
Wasn't sure about exactly which logical condition would be best to check for. Important thing is that the train call to pomegranate doesn't crash.